js/reference.js

```diff
@@ -783,15 +760,41 @@ var ERMrest = (function(module) {
      * this data was acquired.
      * @param {!Object} data The unprocessed tuple of data returned from ERMrest.
      */
-    function Tuple(reference, data) {
-        this._ref = reference;
+    function Tuple(pageReference, data) {
+        this._pageRef = pageReference;
         this._data = data;
     }
 
     Tuple.prototype = {
         constructor: Tuple,
 
         /**
+         * This is the reference of the Tuple
+         * @returns {ERMrest.Reference|*} reference of the Tuple
+         */
+        get reference() {
+
+            if (this._ref === undefined) {
+                this._ref = _referenceCopy(this._pageRef);
+
+                // update its location by adding the tuple’s key filter to the URI
+                // don't keep any modifiers
+                var uri = this._pageRef._location.service + "/catalog/" + this._pageRef._location.catalog + "/" +
+                    this._pageRef._location.api + "/" + this._pageRef._table.schema.name + ":" + this._pageRef._table.name + "/";
+                for (var k = 0; k < this._pageRef._shortestKey.length; k++) {
+                    var col = this._pageRef._shortestKey[k].name;
+                    if (k === 0) {
+                        uri = uri + module._fixedEncodeURIComponent(col) + "=" + module._fixedEncodeURIComponent(this._data[col]);
+                    } else {
+                        uri = uri + "&" + module._fixedEncodeURIComponent(col) + "=" + module._fixedEncodeURIComponent(this._data[col]);
+                    }
+                }
+                this._ref._location = new module.Location(uri);
```

Comment:
Another case for `module._parse(uri)` instead of needing the Location constructor.